### DISCUSSION TITLE
Why do we use strongly type Ids instead of simple primitive types?

### DISCUSSION THREAD

**Discussion by mehdihadeli** [kgrzybek/modular-monolith-with-ddd#176] [Q&A] [answered]
Hi,
Why in this project for `ID` property of our Entities you used `strongly typed` IDs instead of simple primitive types?

**Comment by pawelpodsiadly** [comment, marked as answer]
A quote from the article linked below
> By using strongly-typed IDs like these we can take full advantage of the C# type system to ensure different concepts cannot be accidentally used interchangeably. Using these types in the core of your domain will help prevent simple bugs like incorrect argument order issues, which can be easy to do, and tricky to spot!

[Using strongly-typed entity IDs to avoid primitive obsession](https://andrewlock.net/using-strongly-typed-entity-ids-to-avoid-primitive-obsession-part-1/)

**Reply by mehdihadeli** [reply]
@pawelpodsiadly Thanks.
